### ISSUE TITLE
Train loss decrease but test increase ? 

### ISSUE THREAD

**Issue by zlheos** [xingyizhou/CenterNet#335]
2019-09-06-05-36: epoch: 4 |loss 0.934375 | hm_loss 0.703657 | wh_loss 0.718979 | off_loss 0.158820 | time 33.716667 | 
2019-09-06-06-10: epoch: 5 |loss 0.915650 | hm_loss 0.687424 | wh_loss 0.710136 | off_loss 0.157213 | time 33.983333 | loss 0.878701 | hm_loss 0.655641 | wh_loss 0.741875 | off_loss 0.148873 | time 19.500000 | 
2019-09-06-07-04: epoch: 6 |loss 0.901647 | hm_loss 0.675190 | wh_loss 0.703902 | off_loss 0.156066 | time 33.966667 | 
2019-09-06-07-38: epoch: 7 |loss 0.889666 | hm_loss 0.665239 | wh_loss 0.694920 | off_loss 0.154935 | time 34.066667 | 
2019-09-06-08-12: epoch: 8 |loss 0.880927 | hm_loss 0.657863 | wh_loss 0.689486 | off_loss 0.154116 | time 34.200000 | 
2019-09-06-08-47: epoch: 9 |loss 0.873890 | hm_loss 0.652165 | wh_loss 0.684874 | off_loss 0.153237 | time 34.366667 | 
2019-09-06-09-21: epoch: 10 |loss 0.867523 | hm_loss 0.646704 | wh_loss 0.681443 | off_loss 0.152675 | time 33.966667 | loss 0.846405 | hm_loss 0.629960 | wh_loss 0.706149 | off_loss 0.145830 | time 19.583333 | 
2019-09-06-10-15: epoch: 11 |loss 0.862302 | hm_loss 0.642243 | wh_loss 0.679519 | off_loss 0.152107 | time 34.150000 | 
2019-09-06-10-49: epoch: 12 |loss 0.858782 | hm_loss 0.639253 | wh_loss 0.678326 | off_loss 0.151697 | time 34.050000 | 
2019-09-06-11-23: epoch: 13 |loss 0.853581 | hm_loss 0.635029 | wh_loss 0.674463 | off_loss 0.151106 | time 34.000000 | 
2019-09-06-11-57: epoch: 14 |loss 0.852138 | hm_loss 0.633552 | wh_loss 0.676922 | off_loss 0.150894 | time 34.033333 | 
2019-09-06-12-31: epoch: 15 |loss 0.849056 | hm_loss 0.630853 | wh_loss 0.675745 | off_loss 0.150629 | time 34.016667 | loss 0.842598 | hm_loss 0.625962 | wh_loss 0.716788 | off_loss 0.144957 | time 19.600000 | 
2019-09-06-13-24: epoch: 16 |loss 0.845896 | hm_loss 0.628124 | wh_loss 0.676030 | off_loss 0.150168 | time 34.016667 | 
2019-09-06-13-59: epoch: 17 |loss 0.843432 | hm_loss 0.626069 | wh_loss 0.672489 | off_loss 0.150114 | time 34.150000 | 
2019-09-06-14-33: epoch: 18 |loss 0.840279 | hm_loss 0.623696 | wh_loss 0.669278 | off_loss 0.149655 | time 34.000000 | 
2019-09-06-15-07: epoch: 19 |loss 0.838199 | hm_loss 0.621964 | wh_loss 0.667318 | off_loss 0.149504 | time 33.933333 | 
2019-09-06-15-41: epoch: 20 |loss 0.835694 | hm_loss 0.619834 | wh_loss 0.667435 | off_loss 0.149117 | time 34.200000 | loss 0.830382 | hm_loss 0.615562 | wh_loss 0.712263 | off_loss 0.143594 | time 19.383333 | 
2019-09-06-16-34: epoch: 21 |loss 0.833725 | hm_loss 0.618315 | wh_loss 0.665664 | off_loss 0.148844 | time 33.950000 | 
2019-09-06-17-08: epoch: 22 |loss 0.831626 | hm_loss 0.616494 | wh_loss 0.664812 | off_loss 0.148650 | time 33.600000 | 
2019-09-06-17-42: epoch: 23 |loss 0.830043 | hm_loss 0.615232 | wh_loss 0.662630 | off_loss 0.148549 | time 33.666667 | 
2019-09-06-18-16: epoch: 24 |loss 0.827255 | hm_loss 0.612906 | wh_loss 0.661154 | off_loss 0.148234 | time 33.983333 | 
2019-09-06-18-50: epoch: 25 |loss 0.827679 | hm_loss 0.613627 | wh_loss 0.660755 | off_loss 0.147977 | time 34.050000 | loss 0.809092 | hm_loss 0.596803 | wh_loss 0.703784 | off_loss 0.141910 | time 20.166667 | 
2019-09-06-19-44: epoch: 26 |loss 0.826894 | hm_loss 0.612689 | wh_loss 0.661917 | off_loss 0.148013 | time 34.166667 | 
2019-09-06-20-18: epoch: 27 |loss 0.825732 | hm_loss 0.611856 | wh_loss 0.661295 | off_loss 0.147746 | time 33.833333 | 
2019-09-06-20-52: epoch: 28 |loss 0.825648 | hm_loss 0.611814 | wh_loss 0.660805 | off_loss 0.147753 | time 33.983333 | 
2019-09-06-21-26: epoch: 29 |loss 0.823970 | hm_loss 0.610586 | wh_loss 0.658035 | off_loss 0.147580 | time 34.283333 | 
2019-09-06-22-00: epoch: 30 |loss 0.822942 | hm_loss 0.609529 | wh_loss 0.660058 | off_loss 0.147408 | time 33.883333 | loss 0.900849 | hm_loss 0.641744 | wh_loss 1.089792 | off_loss 0.150126 | time 19.750000 | 
2019-09-06-22-54: epoch: 31 |loss 0.823029 | hm_loss 0.609602 | wh_loss 0.658855 | off_loss 0.147542 | time 33.916667 | 
2019-09-06-23-28: epoch: 32 |loss 0.822684 | hm_loss 0.609396 | wh_loss 0.658925 | off_loss 0.147396 | time 34.183333 | 
2019-09-07-00-02: epoch: 33 |loss 0.825190 | hm_loss 0.611106 | wh_loss 0.665290 | off_loss 0.147555 | time 34.050000 | 
2019-09-07-00-36: epoch: 34 |loss 0.825736 | hm_loss 0.611650 | wh_loss 0.664917 | off_loss 0.147594 | time 34.066667 | 
2019-09-07-01-10: epoch: 35 |loss 0.823402 | hm_loss 0.609731 | wh_loss 0.663687 | off_loss 0.147302 | time 33.766667 | loss 1.746833 | hm_loss 1.296085 | wh_loss 2.735421 | off_loss 0.177206 | time 19.950000 | 
2019-09-07-02-03: epoch: 36 |loss 0.823220 | hm_loss 0.609004 | wh_loss 0.669513 | off_loss 0.147265 | time 33.500000 | 
2019-09-07-02-38: epoch: 37 |loss 0.822510 | hm_loss 0.608780 | wh_loss 0.665023 | off_loss 0.147227 | time 34.083333 | 
2019-09-07-03-12: epoch: 38 |loss 0.819813 | hm_loss 0.606991 | wh_loss 0.659043 | off_loss 0.146918 | time 33.883333 | 
2019-09-07-03-45: epoch: 39 |loss 0.822166 | hm_loss 0.608184 | wh_loss 0.669371 | off_loss 0.147044 | time 33.883333 | 
2019-09-07-04-19: epoch: 40 |loss 0.820032 | hm_loss 0.606789 | wh_loss 0.664198 | off_loss 0.146823 | time 34.000000 | loss 2.121127 | hm_loss 1.463624 | wh_loss 4.884866 | off_loss 0.169016 | time 19.600000 | 
2019-09-07-05-13: epoch: 41 |loss 0.820232 | hm_loss 0.606886 | wh_loss 0.665751 | off_loss 0.146771 | time 33.816667 | 
2019-09-07-05-47: epoch: 42 |loss 0.819603 | hm_loss 0.606404 | wh_loss 0.664309 | off_loss 0.146768 | time 34.350000 | 
2019-09-07-06-21: epoch: 43 |loss 0.820090 | hm_loss 0.606722 | wh_loss 0.667766 | off_loss 0.146591 | time 33.950000 | 
2019-09-07-06-55: epoch: 44 |loss 0.822297 | hm_loss 0.607917 | wh_loss 0.676611 | off_loss 0.146719 | time 34.066667 | 
2019-09-07-07-29: epoch: 45 |loss 0.817926 | hm_loss 0.605159 | wh_loss 0.664453 | off_loss 0.146321 | time 33.833333 | loss 10.544018 | hm_loss 8.738917 | wh_loss 15.546956 | off_loss 0.250406 | time 19.233333 | 
2019-09-07-08-22: epoch: 46 |loss 0.816823 | hm_loss 0.604771 | wh_loss 0.658960 | off_loss 0.146156 | time 33.283333 | 
2019-09-07-08-56: epoch: 47 |loss 0.816001 | hm_loss 0.603681 | wh_loss 0.660876 | off_loss 0.146232 | time 34.100000 | 
2019-09-07-09-30: epoch: 48 |loss 0.814941 | hm_loss 0.602982 | wh_loss 0.659516 | off_loss 0.146007 | time 34.100000 | 
2019-09-07-10-04: epoch: 49 |loss 0.821963 | hm_loss 0.607241 | wh_loss 0.683991 | off_loss 0.146322 | time 34.033333 | 
2019-09-07-10-38: epoch: 50 |loss 0.813217 | hm_loss 0.601726 | wh_loss 0.656087 | off_loss 0.145882 | time 34.166667 | loss 8.464491 | hm_loss 7.497729 | wh_loss 7.626050 | off_loss 0.204157 | time 19.533333 | 
2019-09-07-11-31: epoch: 51 |loss 0.814883 | hm_loss 0.602931 | wh_loss 0.660642 | off_loss 0.145888 | time 33.216667 | 
2019-09-07-12-05: epoch: 52 |loss 0.814183 | hm_loss 0.602564 | wh_loss 0.658414 | off_loss 0.145778 | time 33.950000 | 
2019-09-07-12-39: epoch: 53 |loss 0.813680 | hm_loss 0.601901 | wh_loss 0.660235 | off_loss 0.145756 | time 33.833333 | 
2019-09-07-13-13: epoch: 54 |loss 0.814149 | hm_loss 0.602437 | wh_loss 0.660559 | off_loss 0.145657 | time 33.916667 | 
2019-09-07-13-47: epoch: 55 |loss 0.817744 | hm_loss 0.604341 | wh_loss 0.675472 | off_loss 0.145856 | time 33.650000 | loss 3.374118 | hm_loss 2.674620 | wh_loss 5.294612 | off_loss 0.170037 | time 19.633333 | 
2019-09-07-14-40: epoch: 56 |loss 0.814406 | hm_loss 0.602691 | wh_loss 0.661157 | off_loss 0.145599 | time 34.000000 | 
2019-09-07-15-15: epoch: 57 |loss 0.821182 | hm_loss 0.606831 | wh_loss 0.684577 | off_loss 0.145894 | time 34.166667 | 
2019-09-07-15-48: epoch: 58 |loss 0.813121 | hm_loss 0.601959 | wh_loss 0.657258 | off_loss 0.145436 | time 33.800000 | 
2019-09-07-16-22: epoch: 59 |loss 0.815097 | hm_loss 0.602876 | wh_loss 0.665114 | off_loss 0.145709 | time 33.866667 | 
2019-09-07-16-56: epoch: 60 |loss 0.814226 | hm_loss 0.602438 | wh_loss 0.664383 | off_loss 0.145350 | time 34.050000 | loss 1.117879 | hm_loss 0.818807 | wh_loss 1.515466 | off_loss 0.147525 | time 19.533333 |

**Comment by xingyizhou** [comment]
Hi, first make sure your data format is correct, by looking at the images with `--debug 2`. It does happen that the validation loss increases as the training goes on, see https://github.com/xingyizhou/CenterNet/issues/148.

**Comment by zlheos** [comment]
I verify annotation, they are right
### 1) step1
I test train_set and val_set by using model_110.pth
there has no objects detected
but epoch 100, the train loss is 0.67
### 2）step2
I test train_set and val_set by using model_25.pth
there has many objects detected 
and epoch 25, the train loss is 0.80

I don't know why ?

**Comment by zlheos** [comment]
**I train with 4GPUS**
they are normal

**I train with 8GPUS**
they are abnormal from epoch 30

**Comment by Beam-wi** [comment]
It seems the model of shock. I have the same problem. I prepare to try warmup for it.

**Comment by xingyizhou** [comment]
The behavior looks like an old pytorch bug discussed [here](https://github.com/xingyizhou/pytorch-pose-hg-3d/issues/16). You can try disabling the cudnn batchnorm as in that issue. However I haven't met this in this project ...

**Comment by zlheos** [comment]
@xingyizhou , I test with cudnn.backends = False, the result is right; thank you !!!

**Comment by zlheos** [comment]
use mokey_patch_bn.py
